File: Nivedita/Linkedlist/Reverse Linked List.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
     ListNode*Next,*prev=NULL;
        
      ListNode*curr=head;
        while(curr!=NULL)
        {
        Next=curr->next;
        curr->next=prev;
        prev=curr;
        curr=Next;
        }
        return prev;
        
        
    }
};

File: Saummya/LINKEDLIST/LEETCODE/Reverse Linked List.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        nodes=[]
        temp=head
        while temp:
            nodes.append(temp)
            temp=temp.next
            
        ans=None
        while nodes:
            node=nodes.pop()
            if not ans:
                ans=node
            node.next=nodes[-1] if nodes else None
            
        return ans
